spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServicePropertySourceLocator.java

```diff
@@ -187,67 +182,38 @@ public void setRestTemplate(RestTemplate restTemplate) {
 		this.restTemplate = restTemplate;
 	}
 
-	private RestTemplate getSecureRestTemplate(ConfigClientProperties client) {
+	private RestTemplate getRestTemplate() {
+		if (this.restTemplate != null) {
+			return this.restTemplate;
+		}
 		SimpleClientHttpRequestFactory requestFactory = new SimpleClientHttpRequestFactory();
 		requestFactory.setReadTimeout((60 * 1000 * 3) + 5000); //TODO 3m5s, make configurable?
 		RestTemplate template = new RestTemplate(requestFactory);
-		String password = client.getPassword();
-		String authorization = client.getAuthorization();
-
-		if (password != null && authorization != null) {
-			throw new IllegalStateException(
-					"You must set either 'password' or 'authorization'");
-		}
-
-		if (password != null) {
-			template.setInterceptors(Arrays.<ClientHttpRequestInterceptor> asList(
-					new BasicAuthorizationInterceptor(client.getUsername(), password)));
-		}
-		else if (authorization != null) {
-			template.setInterceptors(Arrays.<ClientHttpRequestInterceptor> asList(
-					new GenericAuthorization(authorization)));
-		}
-
 		return template;
 	}
 
-	private static class BasicAuthorizationInterceptor implements
```

Comment:
I don't think your approach is necessarily wrong, it would just be less churn in the code if you kept the interceptor and generalized it to add the new headers as well.